src/cli/standalone-daemon.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+const debug = require('debug')
+
+const IPFS = require('../core')
+const HttpApi = require('../http')
+const WStar = require('libp2p-webrtc-star')
+const TCP = require('libp2p-tcp')
+const MulticastDNS = require('libp2p-mdns')
+const WS = require('libp2p-websockets')
+const Bootstrap = require('libp2p-bootstrap')
+
+class StandaloneDaemon {
+  constructor (options) {
+    this._options = options || {}
+    this._log = debug('ipfs:daemon')
+    this._log.error = debug('ipfs:daemon:error')
+
+    if (process.env.IPFS_MONITORING) {
+      // Setup debug metrics collection
+      const prometheusClient = require('prom-client')
+      const prometheusGcStats = require('prometheus-gc-stats')
+      const collectDefaultMetrics = prometheusClient.collectDefaultMetrics
+      collectDefaultMetrics({ timeout: 5000 })
+      prometheusGcStats(prometheusClient.register)()
+    }
+  }
+
+  async start () {
+    this._log('starting')
+
+    const libp2p = { modules: {} }
+
+    // Attempt to use any of the WebRTC versions available globally
+    let electronWebRTC
+    let wrtc
+    try {
+      electronWebRTC = require('electron-webrtc')()
+    } catch (err) {
+      this._log('failed to load optional electron-webrtc dependency')
+    }
+    try {
+      wrtc = require('wrtc')
+    } catch (err) {
+      this._log('failed to load optional webrtc dependency')
+    }
+
+    if (wrtc || electronWebRTC) {
+      const using = wrtc ? 'wrtc' : 'electron-webrtc'
+      this._log(`Using ${using} for webrtc support`)
+      const wstar = new WStar({ wrtc: (wrtc || electronWebRTC) })
+      libp2p.modules.transport = [TCP, WS, wstar]
+      libp2p.modules.peerDiscovery = [MulticastDNS, Bootstrap, wstar.discovery]
+    }
+
+    // start the daemon
+    const ipfsOpts = Object.assign({ init: false }, this._options, { start: true, libp2p })
+    const ipfs = new IPFS(ipfsOpts)
+
+    await new Promise((resolve, reject) => {
+      ipfs.once('error', err => {
+        this._log('error starting core', err)
+        err.code = 'ENOENT'
+        reject(err)
+      })
+      ipfs.once('start', resolve)
+    })
```

Comment:
Argh, any chance you can help @pkafei writing a test for https://github.com/ipfs/js-ipfs/pull/1878 so we don't need to do this?

(completely optional and I won't block merging this!)

Reply:
Can we pair program and work on the test together?

Reply:
@alanshaw I'd prefer to merge this PR as-is and improve it later.

@pkafei Sure! I am booked today, but feel free to grab time slot next week: https://calendly.com/pl-lidel/ :)